### PR TITLE
ETK: Add function for FSE eligibility

### PR DESCRIPTION
**Pull request by creativecoder**
#### Changes proposed in this Pull Request

* Adds the function `\A8C\FSE\is_core_fse_eligible` for determining eligibility for the FSE Beta.

#### Testing instructions
 
Note: requires a wpcom sandbox with D67076-code applied and a local development site with Jetpack connected to WP.com for testing

Use run `yarn run dev --sync` from the `apps/editing-toolkit` folder to sync the changes to your sandbox

See D67076-code for more detailed testing instructions

Related to #56373

### CODE REVIEW

**Comment by mattwiebe** [comment]
Why are we doing this? The whole point, I thought, of using a blog sticker for eligibility was that it was nice and simple to have in a wpcom endpoint, regardless of Simple vs Atomic. We only need to know this in a Calypso context, so we shouldn't even need this?

**Comment by creativecoder** [comment]
> Why are we doing this? The whole point, I thought, of using a blog sticker for eligibility was that it was nice and simple to have in a wpcom endpoint, regardless of Simple vs Atomic.

I don't think there's anything here that negates that. The blog sticker continues to control the FSE Beta eligiblity. For Simple sites, wpcom knows to short-circuit the request and calls the endpoint directly (D67076-code), which just checks for the sticker. For Atomic sites, they need to make an http request to wpcom because they don't have direct access to the blog sticker.

It seems best to have Beta eligibility inside a lib function, rather than directly testing a sticker in multiple places. This will make it easier to change eligibility (e.g. when everyone becomes "eligible"). Our lib code for FSE functionality is shipped to both Simple and Atomic sites, so the change here works in both contexts. wpcom code can be updated to use `is_core_fse_eligible()` in the places we're currently checking for a sticker (Atomic sites can use that function too, should we find a need). The endpoint becomes the source of truth, when we need to make any updates for determining Beta eligibility.

I can see how it seems a little backwards (usually the endpoint consumes the library, not the other way around), but I think this works well for keeping things contained in the same place (`\A8C\FSE`) and not needing branching logic depending upon the environment.

**Comment by creativecoder** [comment]
Putting this on hold for now while finishing D65498-code. I'll resume this PR if needed, while looking at where we need to know if a site is FSE Beta eligible.

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-universal-themes/index.php [hunk 1]
```diff
@@ -9,11 +9,45 @@
...
  */
 define( 'ACTIVATE_FSE_OPTION_NAME', 'wpcom_is_fse_activated' );
 
+/**
+ * Checks if site is eligible for Core FSE
+ *
+ * As of 2021, this is only sites for existing users who explicitly
+ * choose to enroll in the WP.com FSE Beta
+ */
+function is_core_fse_eligible() {
```
Comment (vindl) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-universal-themes/index.php:25]: Similar to the comment in the diff, I think we can start dropping the `core` part when naming FSE things.

**Review by vindl — COMMENTED** [review]
> It seems best to have Beta eligibility inside a lib function, rather than directly testing a sticker in multiple places. This will make it easier to change eligibility (e.g. when everyone becomes "eligible").

I agree with this. If we ever want to refactor eligibility check, it's better to have it encapsulated in one place, instead of chasing all of the sticker usages.

> I can see how it seems a little backwards (usually the endpoint consumes the library, not the other way around), but I think this works well for keeping things contained in the same place (\A8C\FSE) and not needing branching logic depending upon the environment.

I found this a bit unexpected too. When this code is executed in WP.com context, we can check for eligibility directly without going through API hoops. That still happens behind the scenes with `wpcom_json_api_request_as_blog` but it's less readable and efficient.

On the other hand, when executing this in Atomic/Jetpack context - do we really need to wait for an API roundtrip? We can't really prevent them from installing FSE themes, so in that sense, they can all be considered eligible by default. Based on this, could we fall back to checking the `is_fse_active` option for Atomic/Jetpack sites?

**Comment by creativecoder** [comment]
We haven't come up with a use case so far for needing to check FSE eligibility outside wpcom, so I'll go ahead and close this, for now. There are wpcom only helper functions (e.g. `is_fse_eligible`) that serve our needs at the moment.

**Pull request closed without merging**